specifyweb/stored_queries/format.py

```diff
@@ -119,11 +119,21 @@ def make_expr(self,
         formatter = fieldNodeAttrib.get('formatter', None)
         aggregator = fieldNodeAttrib.get('aggregator', None)
         next_table_name = formatter_field_spec.table.name
+
+        # preserve sql table alias
+        formatter_field_spec = formatter_field_spec._replace(root_sql_table=orm_table)
+
         if formatter_field_spec.is_relationship():
             if previous_tables is not None and next_table_name in [table_name for table_name, _ in previous_tables]:
-                return (query, literal(
-                    _text(f"<Cycle Detected.>: {'->'.join([*[str(_) for _ in previous_tables], next_table_name])}")),
-                        formatter_field_spec)
+                if query.detect_cycles:
+                    formatted_literal = literal(_text(f"<Cycle Detected.>: {'->'.join([*[str(_) for _ in previous_tables], next_table_name])}"))
+                else:
+                    formatted_literal = literal(
+                        _text("<Invalid record formatter detected>") if formatter_field_spec.get_field().type == 'many-to-one' else
+                        _text("<Invalid record aggregator detected>")
+                    )
```

Comment:
I don't think we should be outputting `Invalid record formatter detected`.

reasons:
1. it's easy to accidentally create a cycle in the UI, and the UI doesn't provide an easy way to detect that it has happened
2. this string is very long

instead, I would suggest you match the front-end behavior and simply short-circuit the output once a cycle is detected (don't output anything/output empty string)

that is going to be the desirable behavior in most cases.
i.e a non-technical user wouldn't understand what a cycle in this context or why it's occurring. and resolving the cycle would force them to create almost identical duplicated formatters (one with a cyclical relationship, and another without)
where as simply silently short-circuiting would make this "just work"

Reply:
After internal discussion, it was decided to keep the text.